src/main/java/com/softwaremill/react/kafka/PropertiesBuilder.java

```diff
@@ -0,0 +1,158 @@
+package com.softwaremill.react.kafka;
+
+import kafka.serializer.*;
+import kafka.utils.VerifiableProperties;
+import scala.collection.JavaConverters;
+import scala.collection.immutable.HashMap;
+
+import java.util.Map;
+
+/**
+ * Builder class wrapping Consumer & Producer properties creation in Java API.
+ */
+public class PropertiesBuilder {
+
+    /**
+     * The Consumer Builder
+     */
+    public static class Consumer {
+
+        private String topic;
+        private String groupId;
+        private Decoder decoder;
+        private String brokerList;
+        private String zooKeeperHost;
+        private scala.collection.immutable.Map<String, String> consumerParams = new HashMap<>();
+
+        public Consumer withTopic(String topic) {
+            this.topic = topic;
+            return this;
+        }
+
+        public Consumer withGroupId(String groupId) {
+            this.groupId = groupId;
+            return this;
+        }
+
+        public Consumer withBrokerList(String brokerList) {
+            this.brokerList = brokerList;
+            return this;
+        }
+
+        public Consumer withZooKeeperHost(String zooKeeperHost) {
+            this.zooKeeperHost = zooKeeperHost;
+            return this;
+        }
+
+        public Consumer withStringDecoder(VerifiableProperties props) {
+            this.decoder = new StringDecoder(props);
+            return this;
+        }
+
+        public Consumer withStringDecoder() {
+            this.decoder = new StringDecoder(null);
+            return this;
+        }
+
+        public Consumer withDefaultDecoder(VerifiableProperties props) {
+            this.decoder = new DefaultDecoder(props);
+            return this;
+        }
+
+        public Consumer withDefaultDecoder() {
+            this.decoder = new DefaultDecoder(null);
+            return this;
+        }
+
+        public Consumer withParams(Map<String, String> params) {
+            this.consumerParams = new HashMap<String, String>().$plus$plus(JavaConverters.mapAsScalaMapConverter(params).asScala());
+            return this;
+        }
+
+        public <C> ConsumerProperties<C> build() {
+            if (brokerList != null && zooKeeperHost != null) {
+                return ConsumerProperties.<C>apply(brokerList, zooKeeperHost, topic, groupId, decoder);
+            }
+            return ConsumerProperties.<C>apply(consumerParams, topic, groupId, decoder);
```

Comment:
personally I would add an `else {}` - or remove the curly brackets around the `if` block?